<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Ivan Osinnii">
<meta name="dcterms.date" content="2024-05-06">

<title>Automated MAIT cell annotation in Lett dataset (blood) using SingleR</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="4-Lett-data-SingleR-blood_files/libs/clipboard/clipboard.min.js"></script>
<script src="4-Lett-data-SingleR-blood_files/libs/quarto-html/quarto.js"></script>
<script src="4-Lett-data-SingleR-blood_files/libs/quarto-html/popper.min.js"></script>
<script src="4-Lett-data-SingleR-blood_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="4-Lett-data-SingleR-blood_files/libs/quarto-html/anchor.min.js"></script>
<link href="4-Lett-data-SingleR-blood_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="4-Lett-data-SingleR-blood_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="4-Lett-data-SingleR-blood_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="4-Lett-data-SingleR-blood_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="4-Lett-data-SingleR-blood_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<style>

      .quarto-title-block .quarto-title-banner h1,
      .quarto-title-block .quarto-title-banner h2,
      .quarto-title-block .quarto-title-banner h3,
      .quarto-title-block .quarto-title-banner h4,
      .quarto-title-block .quarto-title-banner h5,
      .quarto-title-block .quarto-title-banner h6
      {
        color: black;
      }

      .quarto-title-block .quarto-title-banner {
        color: black;
background: #f0f3f5;
      }
</style>


<link rel="stylesheet" href="style.css">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Automated MAIT cell annotation in Lett dataset (blood) using SingleR</h1>
            <p class="subtitle lead">scRNA-seq workflow series part5</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Ivan Osinnii </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">May 6, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#load-data-and-dependencies" id="toc-load-data-and-dependencies" class="nav-link" data-scroll-target="#load-data-and-dependencies"><span class="header-section-number">2</span> Load data and dependencies</a></li>
  <li><a href="#analysis-with-singler-and-monacoimmunedata-reference" id="toc-analysis-with-singler-and-monacoimmunedata-reference" class="nav-link" data-scroll-target="#analysis-with-singler-and-monacoimmunedata-reference"><span class="header-section-number">3</span> Analysis with SingleR and MonacoImmuneData reference</a>
  <ul class="collapse">
  <li><a href="#make-an-annotation-prediction-model-using-singler-function" id="toc-make-an-annotation-prediction-model-using-singler-function" class="nav-link" data-scroll-target="#make-an-annotation-prediction-model-using-singler-function"><span class="header-section-number">3.0.1</span> Make an annotation prediction model using SingleR function</a></li>
  <li><a href="#validate-annotation-using-quality-control-plots" id="toc-validate-annotation-using-quality-control-plots" class="nav-link" data-scroll-target="#validate-annotation-using-quality-control-plots"><span class="header-section-number">3.0.2</span> Validate annotation using quality control plots</a></li>
  <li><a href="#we-make-the-projection-of-all-monacoimmunedata-cell-types-on-lett-dataset" id="toc-we-make-the-projection-of-all-monacoimmunedata-cell-types-on-lett-dataset" class="nav-link" data-scroll-target="#we-make-the-projection-of-all-monacoimmunedata-cell-types-on-lett-dataset"><span class="header-section-number">3.0.3</span> We make the projection of all MonacoImmuneData cell types on Lett dataset</a></li>
  <li><a href="#next-we-want-to-do-some-adjusted-vusualizations-to-focus-only-on-mait-cells" id="toc-next-we-want-to-do-some-adjusted-vusualizations-to-focus-only-on-mait-cells" class="nav-link" data-scroll-target="#next-we-want-to-do-some-adjusted-vusualizations-to-focus-only-on-mait-cells"><span class="header-section-number">3.0.4</span> Next we want to do some adjusted vusualizations to focus only on MAIT cells</a></li>
  <li><a href="#mait-cell-counts-compared-to-total-t-cells" id="toc-mait-cell-counts-compared-to-total-t-cells" class="nav-link" data-scroll-target="#mait-cell-counts-compared-to-total-t-cells"><span class="header-section-number">3.0.5</span> MAIT cell counts compared to total T cells</a></li>
  <li><a href="#conclusion-4.3-mait-cells-out-of-all-t-cells-in-blood" id="toc-conclusion-4.3-mait-cells-out-of-all-t-cells-in-blood" class="nav-link" data-scroll-target="#conclusion-4.3-mait-cells-out-of-all-t-cells-in-blood"><span class="header-section-number">3.0.6</span> Conclusion: 4.3 % MAIT cells out of all T cells in blood</a></li>
  </ul></li>
  <li><a href="#analysis-with-singler-and-garner-et-al.-reference" id="toc-analysis-with-singler-and-garner-et-al.-reference" class="nav-link" data-scroll-target="#analysis-with-singler-and-garner-et-al.-reference"><span class="header-section-number">4</span> Analysis with SingleR and Garner et al.&nbsp;reference</a>
  <ul class="collapse">
  <li><a href="#extract-count-matrix-from-seurat-object-for-ref-data-and-adjust-both-query-and-reference" id="toc-extract-count-matrix-from-seurat-object-for-ref-data-and-adjust-both-query-and-reference" class="nav-link" data-scroll-target="#extract-count-matrix-from-seurat-object-for-ref-data-and-adjust-both-query-and-reference"><span class="header-section-number">4.0.1</span> Extract count matrix from Seurat object for ref data and adjust both query and reference</a></li>
  <li><a href="#lets-construct-reference-object-from-garner-dataset" id="toc-lets-construct-reference-object-from-garner-dataset" class="nav-link" data-scroll-target="#lets-construct-reference-object-from-garner-dataset"><span class="header-section-number">4.0.2</span> Let’s construct reference object from Garner dataset</a></li>
  <li><a href="#we-make-second-mait-annotation-prediction-model-based-on-garner-reference" id="toc-we-make-second-mait-annotation-prediction-model-based-on-garner-reference" class="nav-link" data-scroll-target="#we-make-second-mait-annotation-prediction-model-based-on-garner-reference"><span class="header-section-number">4.0.3</span> We make second MAIT annotation prediction model based on Garner reference</a></li>
  <li><a href="#annotation-qc-plots" id="toc-annotation-qc-plots" class="nav-link" data-scroll-target="#annotation-qc-plots"><span class="header-section-number">4.0.4</span> Annotation “QC” plots</a></li>
  <li><a href="#we-make-umap-projection-of-garner-et-al.-cell-types-on-lett-dataset" id="toc-we-make-umap-projection-of-garner-et-al.-cell-types-on-lett-dataset" class="nav-link" data-scroll-target="#we-make-umap-projection-of-garner-et-al.-cell-types-on-lett-dataset"><span class="header-section-number">4.0.5</span> We make UMAP projection of Garner et al.&nbsp;cell types on Lett dataset</a></li>
  <li><a href="#calculating-mait-cell-number-and-ratio-to-all-t-cells" id="toc-calculating-mait-cell-number-and-ratio-to-all-t-cells" class="nav-link" data-scroll-target="#calculating-mait-cell-number-and-ratio-to-all-t-cells"><span class="header-section-number">4.0.6</span> Calculating MAIT cell number and ratio to all T cells</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">4.0.7</span> Conclusion</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">




<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>This is a Quarto document which nicely combines both R code, its results and text explanation to enable an easy and interactive access for both learners, readers and supervisors to such analyses. To learn more about Quarto see <a href="https://quarto.org" class="uri">https://quarto.org</a>. By default all code lines are blended, but you can show them by clicking on the code button.</p>
</div>
</div>
</div>
<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>In previous analyses we prepared our datasets in terms of data normalization and UMAP projection. We also tried a simple case of MAIT cell annotation taking well-annotated Garner et al.&nbsp;dataset as a query and MonacoImmuneData database as reference.</p>
<p>Here we take preprocessed data: both Lett experiment (blood cell subset) and Garner et al.&nbsp;dataset preprocessed and filtered (for blood tissue and experiment 2). And we try to:<br>
1) Project MonacoImmuneData MAIT cell labels on Lett dataset;<br>
2) Project Garner et al.&nbsp;dataset MAIT cell label on Lett dataset</p>
<p>In this way we want to be sure that if one of the methods shows some result - it is not due to the batch effect and other types of artefacts, or at least, it is not completely biased towards batch effect or whatever else - donor variability etc.</p>
<p>Also, we want to make exact counts of MAIT cells in Lett dataset.</p>
<p>We start with loading data</p>
</section>
<section id="load-data-and-dependencies" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Load data and dependencies</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleR)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(uwot)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(celldex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>lett.sce <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"./../input/Lett.sce.blood.gene.symbol_analyzed.rds"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>lett.sce</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>class: SingleCellExperiment 
dim: 36169 8365 
metadata(3): Samples Samples Samples
assays(5): counts genefull spliced unspliced logcounts
rownames(36169): PLCXD1 LINC00685 ... '' ''
rowData names(12): GENEID SYMBOL ... max_prop_ambient topHVG
colnames(8365): BSSE_QGF_176994-AAACCCAAGATGAATC
  BSSE_QGF_176994-AAACCCAAGGAGCTGT ... BSSE_QGF_176994-TTTGTTGGTTCTTGCC
  BSSE_QGF_176994-TTTGTTGTCGTGGAAG
colData names(23): Barcode SampleName ... label MonImmCell
reducedDimNames(7): PCA TSNE ... UMAP UMAP.2
mainExpName: RNA
altExpNames(0):</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>lett.sce<span class="sc">$</span>cellReferences_celldex_MonacoImmuneData <span class="ot">&lt;-</span> lett.sce<span class="sc">@</span>colData<span class="sc">@</span>listData[[<span class="st">"cellReferences_celldex_MonacoImmuneData"</span>]]<span class="sc">@</span>listData[[<span class="st">"labels"</span>]]</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUMAP</span>(lett.sce, <span class="at">colour_by =</span> <span class="st">'cellReferences_celldex_MonacoImmuneData'</span>,<span class="at">point_size =</span> <span class="fl">1.0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="4-Lett-data-SingleR-blood_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>This UMAP was loaded into the Lett dataset by R. Ivanek and it shows very general immune cell types like T cells, NK cells, B cells. We see that our cells are mainly composed of these types in mentioned order…</p>
</section>
<section id="analysis-with-singler-and-monacoimmunedata-reference" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Analysis with SingleR and MonacoImmuneData reference</h1>
<p>Extract count matrix from sce object for query data and load reference</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>query <span class="ot">&lt;-</span> <span class="fu">assay</span>(lett.sce) </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">&lt;-</span> celldex<span class="sc">::</span><span class="fu">MonacoImmuneData</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="make-an-annotation-prediction-model-using-singler-function" class="level3" data-number="3.0.1">
<h3 data-number="3.0.1" class="anchored" data-anchor-id="make-an-annotation-prediction-model-using-singler-function"><span class="header-section-number">3.0.1</span> Make an annotation prediction model using SingleR function</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>pred <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> query, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.fine)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>pred.Tcell <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> query, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="validate-annotation-using-quality-control-plots" class="level3" data-number="3.0.2">
<h3 data-number="3.0.2" class="anchored" data-anchor-id="validate-annotation-using-quality-control-plots"><span class="header-section-number">3.0.2</span> Validate annotation using quality control plots</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pred<span class="sc">$</span>scores[<span class="dv">7</span><span class="sc">:</span><span class="dv">9</span>,<span class="dv">7</span><span class="sc">:</span><span class="dv">9</span>] </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     Low-density basophils Low-density neutrophils MAIT cells
[1,]             0.1929253               0.1821794  0.3575172
[2,]             0.1470251               0.1414905  0.2950138
[3,]             0.1383138               0.1303763  0.3098028</code></pre>
</div>
</div>
<p>Just scores for each cell type</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotScoreHeatmap</span>(pred)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="4-Lett-data-SingleR-blood_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Now these scores are plotted on heatmap and we see a small MAIT portion annotated</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDeltaDistribution</span>(pred)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="4-Lett-data-SingleR-blood_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>This complex graph shows the rate of false-positives. Nothing important to mention so far.</p>
</section>
<section id="we-make-the-projection-of-all-monacoimmunedata-cell-types-on-lett-dataset" class="level3" data-number="3.0.3">
<h3 data-number="3.0.3" class="anchored" data-anchor-id="we-make-the-projection-of-all-monacoimmunedata-cell-types-on-lett-dataset"><span class="header-section-number">3.0.3</span> We make the projection of all MonacoImmuneData cell types on Lett dataset</h3>
<p>UMAP plot showing all 29 fine cell types including MAIT projected on our dataset. Unfortunately the legend is so small that we can not clearly see how MAITs are designated</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>lett.sce<span class="sc">$</span>singleR.labels.fine <span class="ot">&lt;-</span> pred<span class="sc">$</span>labels[<span class="fu">match</span>(<span class="fu">rownames</span>(lett.sce<span class="sc">@</span>colData),</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pred))] </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUMAP</span>(lett.sce, <span class="at">colour_by =</span> <span class="st">'singleR.labels.fine'</span>,</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="at">point_size =</span> <span class="fl">1.0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="4-Lett-data-SingleR-blood_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>That is why we will produce additional plots below.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="at">Assigned=</span>pred<span class="sc">$</span>labels, <span class="at">Clusters=</span>lett.sce<span class="sc">$</span>cellReferences_celldex_MonacoImmuneData) </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">log10</span>(tab<span class="sc">+</span><span class="dv">10</span>), <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">'white'</span>,<span class="st">'blue'</span>))(<span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="4-Lett-data-SingleR-blood_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>This is another “heatmap” showing the relation of MonacoImmuneData main clusters already contained in the analysis (done by R. Ivanek) and the newly projected detailed labels. We can see that MAIT cells land somewhere within T cell cluster together with many other T cell subtypes</p>
</section>
<section id="next-we-want-to-do-some-adjusted-vusualizations-to-focus-only-on-mait-cells" class="level3" data-number="3.0.4">
<h3 data-number="3.0.4" class="anchored" data-anchor-id="next-we-want-to-do-some-adjusted-vusualizations-to-focus-only-on-mait-cells"><span class="header-section-number">3.0.4</span> Next we want to do some adjusted vusualizations to focus only on MAIT cells</h3>
<p>We use ifelse function to create a new label with only MAIT cells and “others” and plot we plot it</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>cell_type_of_interest <span class="ot">&lt;-</span> <span class="st">"MAIT cells"</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>MAIT_cells <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(lett.sce<span class="sc">@</span>colData<span class="sc">@</span>listData[[<span class="st">"singleR.labels.fine"</span>]] <span class="sc">==</span> <span class="st">"MAIT cells"</span>, <span class="st">"MAIT cells"</span>, <span class="st">"All other"</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(lett.sce)<span class="sc">$</span>MAIT_cells <span class="ot">&lt;-</span> <span class="fu">factor</span>(MAIT_cells)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"MAIT cells"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"All other"</span> <span class="ot">=</span> <span class="st">"lightgrey"</span>)</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotReducedDim</span>(lett.sce, <span class="st">"UMAP"</span>, <span class="at">colour_by =</span> <span class="st">"MAIT_cells"</span>, <span class="at">point_size =</span> <span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="4-Lett-data-SingleR-blood_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="864"></p>
</div>
</div>
<p>So we see our MAIT cells very nicely among others.</p>
</section>
<section id="mait-cell-counts-compared-to-total-t-cells" class="level3" data-number="3.0.5">
<h3 data-number="3.0.5" class="anchored" data-anchor-id="mait-cell-counts-compared-to-total-t-cells"><span class="header-section-number">3.0.5</span> MAIT cell counts compared to total T cells</h3>
<p>This is achieved by next quite complex code section.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Let's count MAIT cells and Other cells</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>MAIT_cell_values <span class="ot">&lt;-</span> <span class="fu">colData</span>(lett.sce)<span class="sc">$</span>MAIT_cells</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>value_counts <span class="ot">&lt;-</span> <span class="fu">table</span>(MAIT_cell_values)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># now let's annotate all T cells</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>lett.sce<span class="sc">$</span>singleR.labels.main <span class="ot">&lt;-</span> pred.Tcell<span class="sc">$</span>labels[<span class="fu">match</span>(<span class="fu">rownames</span>(lett.sce<span class="sc">@</span>colData),</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pred.Tcell))] </span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtering out only "T cells"</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>t_cell_values <span class="ot">&lt;-</span> lett.sce<span class="sc">$</span>singleR.labels.main</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>is_t_cell <span class="ot">&lt;-</span> t_cell_values <span class="sc">==</span> <span class="st">"T cells"</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>is_t_cell2 <span class="ot">&lt;-</span> t_cell_values <span class="sc">==</span> <span class="st">"CD4+ T cells"</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>is_t_cell3 <span class="ot">&lt;-</span> t_cell_values <span class="sc">==</span> <span class="st">"CD8+ T cells"</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>t_cell_counts <span class="ot">&lt;-</span> <span class="fu">sum</span>(is_t_cell) <span class="sc">+</span> <span class="fu">sum</span>(is_t_cell2) <span class="sc">+</span> <span class="fu">sum</span>(is_t_cell3) <span class="co"># Count the number of "T cells"</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>percent1 <span class="ot">&lt;-</span> <span class="fu">round</span>((value_counts[<span class="dv">2</span>] <span class="sc">/</span>t_cell_counts <span class="sc">*</span> <span class="dv">100</span>),<span class="dv">1</span>)</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">"Number of MAIT cells equals "</span>, value_counts[<span class="dv">2</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Number of MAIT cells equals 188</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">"Number of all T cells equals "</span>, t_cell_counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Number of all T cells equals 4333</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">"Persentage of MAIT cells out of all T cells found with this method of annotation equals "</span>, percent1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Persentage of MAIT cells out of all T cells found with this method of annotation equals 4.3</code></pre>
</div>
</div>
</section>
<section id="conclusion-4.3-mait-cells-out-of-all-t-cells-in-blood" class="level3" data-number="3.0.6">
<h3 data-number="3.0.6" class="anchored" data-anchor-id="conclusion-4.3-mait-cells-out-of-all-t-cells-in-blood"><span class="header-section-number">3.0.6</span> Conclusion: 4.3 % MAIT cells out of all T cells in blood</h3>
</section>
</section>
<section id="analysis-with-singler-and-garner-et-al.-reference" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Analysis with SingleR and Garner et al.&nbsp;reference</h1>
<p>Let’s load the new reference data (blood subset) which we saved from previous Seurat analysis</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>ref2 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"./../output/Garner.seurat.2exp.blood.mait_analyzed.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>This command would be needed to convert reference from Seurat to SCE, but we can avoid it</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">#sceasy::convertFormat(Gar.sce,</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># from = "seurat", to = "sce",</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># outFile = "Gar_sce.rds")</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Gar_sce is garner.blood.withTrm converted into sce</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Gar.sce &lt;- readRDS("Gar_sce.rds")</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Gar.sce</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="extract-count-matrix-from-seurat-object-for-ref-data-and-adjust-both-query-and-reference" class="level3" data-number="4.0.1">
<h3 data-number="4.0.1" class="anchored" data-anchor-id="extract-count-matrix-from-seurat-object-for-ref-data-and-adjust-both-query-and-reference"><span class="header-section-number">4.0.1</span> Extract count matrix from Seurat object for ref data and adjust both query and reference</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>ref2.counts <span class="ot">&lt;-</span>  <span class="fu">GetAssayData</span>(ref2, <span class="at">layer =</span> <span class="st">"data"</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(lett.sce), <span class="fu">rownames</span>(ref2))</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(common_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 15243</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset lett.sce to include only common genes</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>lett.sce <span class="ot">&lt;-</span> lett.sce[common_genes, ]</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>ref2 <span class="ot">&lt;-</span> ref2[common_genes, ]</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>ref2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
15243 features across 40973 samples within 1 assay 
Active assay: RNA (15243 features, 1725 variable features)
 3 layers present: counts, data, scale.data
 2 dimensional reductions calculated: pca, umap</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>query <span class="ot">&lt;-</span> <span class="fu">assay</span>(lett.sce)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="lets-construct-reference-object-from-garner-dataset" class="level3" data-number="4.0.2">
<h3 data-number="4.0.2" class="anchored" data-anchor-id="lets-construct-reference-object-from-garner-dataset"><span class="header-section-number">4.0.2</span> Let’s construct reference object from Garner dataset</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>cell_metadata <span class="ot">&lt;-</span> ref2<span class="sc">$</span>cell_type</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>metadata_df <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(<span class="at">cell_type =</span> cell_metadata)</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>ref_se_object <span class="ot">&lt;-</span> <span class="fu">SummarizedExperiment</span>(<span class="at">assays =</span> ref2.counts,</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> cell_metadata)</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>ref_se_object<span class="sc">$</span>cell_type <span class="ot">&lt;-</span> ref_se_object<span class="sc">$</span>X</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>ref_se_object<span class="sc">@</span>assays<span class="sc">@</span>data<span class="sc">@</span>listData[[<span class="st">"logcounts"</span>]] <span class="ot">&lt;-</span> ref_se_object<span class="sc">@</span>assays<span class="sc">@</span>data<span class="sc">@</span>listData[[<span class="dv">1</span>]]</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>ref_se_object</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>class: SummarizedExperiment 
dim: 19289 40973 
metadata(0):
assays(2): '' logcounts
rownames(19289): AL627309.1 FAM87B ... AC233755.1 AC240274.1
rowData names(0):
colnames(40973): 1_AAACCTGAGCAGCGTA 1_AAACCTGCATTCTCAT ...
  8_TTTGTCAAGTCGAGTG 8_TTTGTCACATATACCG
colData names(2): X cell_type</code></pre>
</div>
</div>
</section>
<section id="we-make-second-mait-annotation-prediction-model-based-on-garner-reference" class="level3" data-number="4.0.3">
<h3 data-number="4.0.3" class="anchored" data-anchor-id="we-make-second-mait-annotation-prediction-model-based-on-garner-reference"><span class="header-section-number">4.0.3</span> We make second MAIT annotation prediction model based on Garner reference</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>pred2 <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> query, <span class="at">ref =</span> ref_se_object, <span class="at">labels =</span> ref_se_object<span class="sc">$</span>cell_type)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="annotation-qc-plots" class="level3" data-number="4.0.4">
<h3 data-number="4.0.4" class="anchored" data-anchor-id="annotation-qc-plots"><span class="header-section-number">4.0.4</span> Annotation “QC” plots</h3>
<p>We see here that MAIT are annotated with very high score, and there is minimal similarity between Tmem and MAIT in our dataset. But “Other” cells have significant similarity with both MAIT and Tmem</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotScoreHeatmap</span>(pred2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="4-Lett-data-SingleR-blood_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>There seems to be no false positives</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDeltaDistribution</span>(pred2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="4-Lett-data-SingleR-blood_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="we-make-umap-projection-of-garner-et-al.-cell-types-on-lett-dataset" class="level3" data-number="4.0.5">
<h3 data-number="4.0.5" class="anchored" data-anchor-id="we-make-umap-projection-of-garner-et-al.-cell-types-on-lett-dataset"><span class="header-section-number">4.0.5</span> We make UMAP projection of Garner et al.&nbsp;cell types on Lett dataset</h3>
<p>It looks simple. We see MAIT likely in the same position as in previous projection model, but they seem to be a little bit less now</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>lett.sce<span class="sc">$</span>label.Garner <span class="ot">&lt;-</span> pred2<span class="sc">$</span>labels[<span class="fu">match</span>(<span class="fu">rownames</span>(lett.sce<span class="sc">@</span>colData), <span class="fu">rownames</span>(pred2))] </span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUMAP</span>(lett.sce, <span class="at">colour_by =</span> <span class="st">'label.Garner'</span>,<span class="at">point_size =</span> <span class="fl">1.0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="4-Lett-data-SingleR-blood_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="864"></p>
</div>
</div>
<p>We use ifelse function to create new labels with only MAIT label and “others”</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">#cell_type_of_interest &lt;- "MAIT cells"</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>MAIT_cells <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(lett.sce<span class="sc">@</span>colData<span class="sc">@</span>listData[[<span class="st">"label.Garner"</span>]] <span class="sc">==</span> <span class="st">"MAIT"</span>, <span class="st">"MAIT"</span>, <span class="st">"All other"</span>)</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(lett.sce)<span class="sc">$</span>MAIT_cells_Garner <span class="ot">&lt;-</span> <span class="fu">factor</span>(MAIT_cells)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"MAIT"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"All other"</span> <span class="ot">=</span> <span class="st">"lightgrey"</span>)</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">plotReducedDim</span>(lett.sce, <span class="st">"UMAP"</span>, <span class="at">colour_by =</span> <span class="st">"MAIT_cells_Garner"</span>, <span class="at">point_size =</span> <span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="4-Lett-data-SingleR-blood_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="864"></p>
</div>
</div>
</section>
<section id="calculating-mait-cell-number-and-ratio-to-all-t-cells" class="level3" data-number="4.0.6">
<h3 data-number="4.0.6" class="anchored" data-anchor-id="calculating-mait-cell-number-and-ratio-to-all-t-cells"><span class="header-section-number">4.0.6</span> Calculating MAIT cell number and ratio to all T cells</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>MAIT_cell_values2 <span class="ot">&lt;-</span> <span class="fu">colData</span>(lett.sce)<span class="sc">$</span>MAIT_cells_Garner</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>value_counts2 <span class="ot">&lt;-</span> <span class="fu">table</span>(MAIT_cell_values2)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>percent2 <span class="ot">&lt;-</span> <span class="fu">round</span>((value_counts2[<span class="dv">2</span>] <span class="sc">/</span>t_cell_counts <span class="sc">*</span> <span class="dv">100</span>),<span class="dv">1</span>)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">"Number of MAIT cells equals "</span>, value_counts2[<span class="dv">2</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Number of MAIT cells equals 475</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">"Number of all T cells equals "</span>, t_cell_counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Number of all T cells equals 4333</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">"Persentage of MAIT cells out of all T cells found with this method of annotation equals "</span>, percent2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Persentage of MAIT cells out of all T cells found with this method of annotation equals 11</code></pre>
</div>
</div>
</section>
<section id="conclusion" class="level3" data-number="4.0.7">
<h3 data-number="4.0.7" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">4.0.7</span> Conclusion</h3>
<p>It seems that we identified the position of our MAIT cells correctly, since two independent dataset projections make very similar spatial result. Using Garner dataset we get significantly more cells identified as MAIT. This is a little bit unexpected ( I would expect similar amounts identified by both methods), because I assume MonacoImmuneData dataset uses also blood MAIT cells for extracting labels ( I could not find more specific information about this reference dataset).</p>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>