<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Ivan Osinnii">
<meta name="dcterms.date" content="2024-06-28">

<title>Manual MAIT and other liver cell annotation using AUCell package and Guillaums dataset</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="6_Guilliams_dataset_analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="6_Guilliams_dataset_analysis_files/libs/quarto-html/quarto.js"></script>
<script src="6_Guilliams_dataset_analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="6_Guilliams_dataset_analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="6_Guilliams_dataset_analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="6_Guilliams_dataset_analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="6_Guilliams_dataset_analysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="6_Guilliams_dataset_analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="6_Guilliams_dataset_analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="6_Guilliams_dataset_analysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<style>

      .quarto-title-block .quarto-title-banner h1,
      .quarto-title-block .quarto-title-banner h2,
      .quarto-title-block .quarto-title-banner h3,
      .quarto-title-block .quarto-title-banner h4,
      .quarto-title-block .quarto-title-banner h5,
      .quarto-title-block .quarto-title-banner h6
      {
        color: black;
      }

      .quarto-title-block .quarto-title-banner {
        color: black;
background: #f0f3f5;
      }
</style>


<link rel="stylesheet" href="style.css">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Manual MAIT and other liver cell annotation using AUCell package and Guillaums dataset</h1>
            <p class="subtitle lead">scRNA-seq workflow series part 6</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Ivan Osinnii </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">June 28, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#load-data-and-dependencies" id="toc-load-data-and-dependencies" class="nav-link" data-scroll-target="#load-data-and-dependencies"><span class="header-section-number">2</span> Load data and dependencies</a>
  <ul class="collapse">
  <li><a href="#load-previously-processed-guillaums-dataset-check-cells-on-umap." id="toc-load-previously-processed-guillaums-dataset-check-cells-on-umap." class="nav-link" data-scroll-target="#load-previously-processed-guillaums-dataset-check-cells-on-umap."><span class="header-section-number">2.0.1</span> Load previously processed Guillaums dataset, check cells on UMAP.</a></li>
  <li><a href="#convert-seurat-object-to-singlecellexperiment" id="toc-convert-seurat-object-to-singlecellexperiment" class="nav-link" data-scroll-target="#convert-seurat-object-to-singlecellexperiment"><span class="header-section-number">2.0.2</span> Convert Seurat object to SingleCellExperiment</a></li>
  <li><a href="#load-the-list-of-mait-and-other-marker-genes." id="toc-load-the-list-of-mait-and-other-marker-genes." class="nav-link" data-scroll-target="#load-the-list-of-mait-and-other-marker-genes."><span class="header-section-number">2.0.3</span> Load the list of MAIT and other marker genes.</a></li>
  </ul></li>
  <li><a href="#analysis" id="toc-analysis" class="nav-link" data-scroll-target="#analysis"><span class="header-section-number">3</span> Analysis</a>
  <ul class="collapse">
  <li><a href="#transform-mait-marker-list-into-geneset-object-using-gseabase-functions" id="toc-transform-mait-marker-list-into-geneset-object-using-gseabase-functions" class="nav-link" data-scroll-target="#transform-mait-marker-list-into-geneset-object-using-gseabase-functions"><span class="header-section-number">3.0.1</span> Transform MAIT marker list into GeneSet object using GSEABase functions</a></li>
  <li><a href="#now-model-cell-type-rankings-using-aucell-package-function" id="toc-now-model-cell-type-rankings-using-aucell-package-function" class="nav-link" data-scroll-target="#now-model-cell-type-rankings-using-aucell-package-function"><span class="header-section-number">3.0.2</span> Now model cell type rankings using AUCell package function</a></li>
  <li><a href="#compare-new-cell-projection-and-native-cell-annotations" id="toc-compare-new-cell-projection-and-native-cell-annotations" class="nav-link" data-scroll-target="#compare-new-cell-projection-and-native-cell-annotations"><span class="header-section-number">3.0.3</span> Compare new cell projection and native cell annotations</a></li>
  <li><a href="#assign-our-new-labels-to-coldata-of-the-sce-object" id="toc-assign-our-new-labels-to-coldata-of-the-sce-object" class="nav-link" data-scroll-target="#assign-our-new-labels-to-coldata-of-the-sce-object"><span class="header-section-number">3.0.4</span> Assign our new labels to colData of the SCE object</a></li>
  <li><a href="#lets-plot-newly-assigned-cell-labels-on-umap" id="toc-lets-plot-newly-assigned-cell-labels-on-umap" class="nav-link" data-scroll-target="#lets-plot-newly-assigned-cell-labels-on-umap"><span class="header-section-number">3.0.5</span> Letâ€™s plot newly assigned cell labels on UMAP</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">3.0.6</span> Conclusion</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">




<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>The dataset is consisting of Liver CD45+ and CD45- cells derived from healthy and obese humans described in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE192740">Guilliams M, Bonnardel J, Haest B, Vanderborght B et al.&nbsp;Spatial proteogenomics reveals distinct and evolutionarily conserved hepatic macrophage niches. Cell 2022</a>.</p>
<p>GSE192740_seuratObj_humanRNAseq.rds.gz file was downloaded, further processed with the Seurat workflow in R and saved as an intermediate .rds file containing native cell annotation, clustering and dim reductions.</p>
<p>Here we apply AUCell package from Bioconductor library in order to find MAIT cells within natively annotated T cells together with other cell types. We are also specifically interested in hepatocytes.</p>
</section>
<section id="load-data-and-dependencies" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Load data and dependencies</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AUCell)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(uwot)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSEABase)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="load-previously-processed-guillaums-dataset-check-cells-on-umap." class="level3" data-number="2.0.1">
<h3 data-number="2.0.1" class="anchored" data-anchor-id="load-previously-processed-guillaums-dataset-check-cells-on-umap."><span class="header-section-number">2.0.1</span> Load previously processed Guillaums dataset, check cells on UMAP.</h3>
<p>We can see many cell types annotated by authors in original publication</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>Gil.seurat <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"./../input/Gil_analyzed.rds"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>Gil.seurat</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
28871 features across 78748 samples within 1 assay 
Active assay: RNA (28871 features, 2000 variable features)
 3 layers present: counts, data, scale.data
 2 dimensional reductions calculated: pca, umap</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(Gil.seurat, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">group.by =</span> <span class="st">"annot"</span>, <span class="at">pt.size =</span> <span class="fl">1.0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="6_Guilliams_dataset_analysis_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="768"></p>
</div>
</div>
</section>
<section id="convert-seurat-object-to-singlecellexperiment" class="level3" data-number="2.0.2">
<h3 data-number="2.0.2" class="anchored" data-anchor-id="convert-seurat-object-to-singlecellexperiment"><span class="header-section-number">2.0.2</span> Convert Seurat object to SingleCellExperiment</h3>
<p>We remove parent objects to save RAM memory for further operations</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>sce_obj <span class="ot">&lt;-</span> <span class="fu">as.SingleCellExperiment</span>(Gil.seurat)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>Gil.sce <span class="ot">&lt;-</span> sce_obj</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(Gil.seurat)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(sce_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="load-the-list-of-mait-and-other-marker-genes." class="level3" data-number="2.0.3">
<h3 data-number="2.0.3" class="anchored" data-anchor-id="load-the-list-of-mait-and-other-marker-genes."><span class="header-section-number">2.0.3</span> Load the list of MAIT and other marker genes.</h3>
<p>MAIT markers were taken from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE238137">this paper</a> and the rest of the markers were taken from <a href="https://www.panglaodb.se/markers.html?cell_type=%27Kupffer%20cells%27">PanglaoDB database</a></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span><span class="fu">read_excel</span>(<span class="st">"./../input/CellMarkerMAITcells.xlsx"</span>, <span class="at">sheet =</span> <span class="dv">2</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="do">## Garner and Pangao cell markers list</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="analysis" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Analysis</h1>
<section id="transform-mait-marker-list-into-geneset-object-using-gseabase-functions" class="level3" data-number="3.0.1">
<h3 data-number="3.0.1" class="anchored" data-anchor-id="transform-mait-marker-list-into-geneset-object-using-gseabase-functions"><span class="header-section-number">3.0.1</span> Transform MAIT marker list into GeneSet object using GSEABase functions</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>all.sets <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(genes), <span class="cf">function</span>(x) {</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">GeneSet</span>(genes[[x]], <span class="at">setName=</span>x)        </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>all.sets <span class="ot">&lt;-</span> <span class="fu">GeneSetCollection</span>(all.sets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="now-model-cell-type-rankings-using-aucell-package-function" class="level3" data-number="3.0.2">
<h3 data-number="3.0.2" class="anchored" data-anchor-id="now-model-cell-type-rankings-using-aucell-package-function"><span class="header-section-number">3.0.2</span> Now model cell type rankings using AUCell package function</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>rankings <span class="ot">&lt;-</span> <span class="fu">AUCell_buildRankings</span>(<span class="fu">counts</span>(Gil.sce),</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">plotStats=</span><span class="cn">FALSE</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="do">### Calculate AUC curves for each cell taking our markers</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>cell.aucs <span class="ot">&lt;-</span> <span class="fu">AUCell_calcAUC</span>(all.sets, rankings)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Genes in the gene sets NOT available in the dataset: 
    MAIT:   3 (7% of 43)
    Tmem:   3 (7% of 43)
    Hepatocyte:     2 (5% of 43)
    Cholangiocytes:     2 (5% of 43)
    Kupffer:    3 (7% of 43)</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">assay</span>(cell.aucs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="compare-new-cell-projection-and-native-cell-annotations" class="level3" data-number="3.0.3">
<h3 data-number="3.0.3" class="anchored" data-anchor-id="compare-new-cell-projection-and-native-cell-annotations"><span class="header-section-number">3.0.3</span> Compare new cell projection and native cell annotations</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>new.labels <span class="ot">&lt;-</span> <span class="fu">colnames</span>(results)[<span class="fu">max.col</span>(results)]</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span> <span class="fu">table</span>(new.labels, Gil.sce<span class="sc">$</span>annot)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                
new.labels       B cells Basophils cDC1s cDC2s Cholangiocytes
  Cholangiocytes       0         0     0     0            102
  Hepatocyte           0         1     0     0            137
  Kupffer              0        15     0     0              1
  MAIT                 3        54     0     0            184
  Tmem               753       101   229   891              6
                
new.labels       Circulating NK/NKT Endothelial cells Fibroblasts Hepatocytes
  Cholangiocytes                  0                 3           4           0
  Hepatocyte                      0               281         295        3795
  Kupffer                         0               161           4           1
  MAIT                           13              1274         508          39
  Tmem                         3469               249         293           5
                
new.labels       Macrophages Mig.cDCs Mono+mono derived cells Neutrophils  pDCs
  Cholangiocytes           0        0                       0          10     0
  Hepatocyte              34        0                       1          16     2
  Kupffer               1615        0                    1704        2237     3
  MAIT                   121        0                     134         961     2
  Tmem                  1475       18                   16162        1614   814
                
new.labels       Plasma cells Resident NK T cells
  Cholangiocytes            0           0       1
  Hepatocyte                2           1       5
  Kupffer                   4           0       9
  MAIT                     11         876   12464
  Tmem                    199        2281   23101</code></pre>
</div>
</div>
<p>We can see that many other cell types got mislabeled as MAIT and Tmem cells. This may be due to the batch effect between reference datasets and current data</p>
</section>
<section id="assign-our-new-labels-to-coldata-of-the-sce-object" class="level3" data-number="3.0.4">
<h3 data-number="3.0.4" class="anchored" data-anchor-id="assign-our-new-labels-to-coldata-of-the-sce-object"><span class="header-section-number">3.0.4</span> Assign our new labels to colData of the SCE object</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>new_labels_df <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(new.labels)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Now assign the DataFrame to colData</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(Gil.sce) <span class="ot">&lt;-</span> new_labels_df</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>Gil.sce<span class="sc">$</span>AUCell_cells <span class="ot">&lt;-</span> Gil.sce<span class="sc">$</span>new.labels</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="lets-plot-newly-assigned-cell-labels-on-umap" class="level3" data-number="3.0.5">
<h3 data-number="3.0.5" class="anchored" data-anchor-id="lets-plot-newly-assigned-cell-labels-on-umap"><span class="header-section-number">3.0.5</span> Letâ€™s plot newly assigned cell labels on UMAP</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUMAP</span>(Gil.sce, <span class="at">colour_by =</span> <span class="st">"AUCell_cells"</span>, <span class="at">point_size =</span> <span class="fl">1.0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="6_Guilliams_dataset_analysis_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="768"></p>
</div>
</div>
</section>
<section id="conclusion" class="level3" data-number="3.0.6">
<h3 data-number="3.0.6" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">3.0.6</span> Conclusion</h3>
<p>Comparing the native cell annotations and new annotations I see a few major problems here. First, T mem cell get spilled over to those previously labeled as Mono-derived and NK cells. Next MAIT cells get imposed on a subset of those previously labeled as circulating NKT/NK cells and some others. Hepatocytes seem to be roughly labeled similarly as in original analysis.</p>
<p>So further analysis is needed to refine these results. Perhaps, one would need to cross-check between gene markers sets we chose and the sets derived from other databases, especially dedicated to immune cells.</p>
<button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#collapseOne">
<p>Session Information</p>
</button>
<div id="collapseOne" class="accordion-collapse collapse">
<div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.3 (2024-02-29)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] readxl_1.4.3                Seurat_5.0.3               
 [3] SeuratObject_5.0.1          sp_2.1-3                   
 [5] GSEABase_1.64.0             graph_1.80.0               
 [7] annotate_1.80.0             XML_3.99-0.16.1            
 [9] AnnotationDbi_1.64.1        uwot_0.1.16                
[11] Matrix_1.6-5                scater_1.30.1              
[13] scuttle_1.12.0              lubridate_1.9.3            
[15] forcats_1.0.0               stringr_1.5.1              
[17] dplyr_1.1.4                 purrr_1.0.2                
[19] readr_2.1.5                 tidyr_1.3.1                
[21] tibble_3.2.1                ggplot2_3.5.0              
[23] tidyverse_2.0.0             AUCell_1.24.0              
[25] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0
[27] Biobase_2.62.0              GenomicRanges_1.54.1       
[29] GenomeInfoDb_1.38.8         IRanges_2.36.0             
[31] S4Vectors_0.40.2            BiocGenerics_0.48.1        
[33] MatrixGenerics_1.14.0       matrixStats_1.2.0          

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22          splines_4.3.3            
  [3] later_1.3.2               bitops_1.0-7             
  [5] cellranger_1.1.0          R.oo_1.26.0              
  [7] polyclip_1.10-6           fastDummies_1.7.3        
  [9] lifecycle_1.0.4           globals_0.16.3           
 [11] lattice_0.22-6            MASS_7.3-60.0.1          
 [13] magrittr_2.0.3            plotly_4.10.4            
 [15] rmarkdown_2.26            yaml_2.3.8               
 [17] httpuv_1.6.15             sctransform_0.4.1        
 [19] spam_2.10-0               spatstat.sparse_3.0-3    
 [21] reticulate_1.35.0         cowplot_1.1.3            
 [23] pbapply_1.7-2             DBI_1.2.2                
 [25] RColorBrewer_1.1-3        abind_1.4-5              
 [27] zlibbioc_1.48.2           Rtsne_0.17               
 [29] R.utils_2.12.3            RCurl_1.98-1.14          
 [31] GenomeInfoDbData_1.2.11   ggrepel_0.9.5            
 [33] irlba_2.3.5.1             spatstat.utils_3.0-4     
 [35] listenv_0.9.1             goftest_1.2-3            
 [37] RSpectra_0.16-1           spatstat.random_3.2-3    
 [39] fitdistrplus_1.1-11       parallelly_1.37.1        
 [41] DelayedMatrixStats_1.24.0 leiden_0.4.3.1           
 [43] codetools_0.2-20          DelayedArray_0.28.0      
 [45] tidyselect_1.2.1          farver_2.1.1             
 [47] ScaledMatrix_1.10.0       viridis_0.6.5            
 [49] spatstat.explore_3.2-7    jsonlite_1.8.8           
 [51] BiocNeighbors_1.20.2      progressr_0.14.0         
 [53] ggridges_0.5.6            survival_3.5-8           
 [55] tools_4.3.3               ica_1.0-3                
 [57] Rcpp_1.0.12               glue_1.7.0               
 [59] gridExtra_2.3             SparseArray_1.2.4        
 [61] xfun_0.43                 withr_3.0.0              
 [63] fastmap_1.1.1             fansi_1.0.6              
 [65] digest_0.6.35             rsvd_1.0.5               
 [67] timechange_0.3.0          R6_2.5.1                 
 [69] mime_0.12                 colorspace_2.1-0         
 [71] scattermore_1.2           tensor_1.5               
 [73] spatstat.data_3.0-4       RSQLite_2.3.6            
 [75] R.methodsS3_1.8.2         utf8_1.2.4               
 [77] generics_0.1.3            data.table_1.15.4        
 [79] httr_1.4.7                htmlwidgets_1.6.4        
 [81] S4Arrays_1.2.1            pkgconfig_2.0.3          
 [83] gtable_0.3.4              blob_1.2.4               
 [85] lmtest_0.9-40             XVector_0.42.0           
 [87] htmltools_0.5.8.1         dotCall64_1.1-1          
 [89] scales_1.3.0              png_0.1-8                
 [91] knitr_1.45                rstudioapi_0.16.0        
 [93] reshape2_1.4.4            tzdb_0.4.0               
 [95] nlme_3.1-164              cachem_1.0.8             
 [97] zoo_1.8-12                KernSmooth_2.23-22       
 [99] parallel_4.3.3            miniUI_0.1.1.1           
[101] vipor_0.4.7               pillar_1.9.0             
[103] grid_4.3.3                vctrs_0.6.5              
[105] RANN_2.6.1                promises_1.2.1           
[107] BiocSingular_1.18.0       beachmat_2.18.1          
[109] xtable_1.8-4              cluster_2.1.6            
[111] beeswarm_0.4.0            evaluate_0.23            
[113] cli_3.6.2                 compiler_4.3.3           
[115] rlang_1.1.3               crayon_1.5.2             
[117] future.apply_1.11.2       labeling_0.4.3           
[119] plyr_1.8.9                ggbeeswarm_0.7.2         
[121] stringi_1.8.3             deldir_2.0-4             
[123] viridisLite_0.4.2         BiocParallel_1.36.0      
[125] munsell_0.5.1             Biostrings_2.70.3        
[127] lazyeval_0.2.2            spatstat.geom_3.2-9      
[129] RcppHNSW_0.6.0            hms_1.1.3                
[131] patchwork_1.2.0           sparseMatrixStats_1.14.0 
[133] bit64_4.0.5               future_1.33.2            
[135] KEGGREST_1.42.0           shiny_1.8.1.1            
[137] ROCR_1.0-11               igraph_2.0.3             
[139] memoise_2.0.1             bit_4.0.5                </code></pre>
</div>
</div>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>